Documentation/SwiftGenKit Contexts/CoreData.md

```diff
@@ -0,0 +1,264 @@
+# Core Data parser
+
+## Input
+
+The core data parser accepts a `xcdatamodeld` (or `xcdatamodel`) file. The parser will load all the configurations, entities, and fetch requests for each model.
+
+## Output
+
+The output context has the following structure:
+
+ - `models`: `Array` — A list of parsed models, with each:
+   - `configurations`: `Dictionary<Array<String>>` — Map of configurations and the corresponding list of entity names
+   - `entities`: `Dictionary` — Map of all the entities in the model, each key being the entity's name:
+     - `name`: `String` - Entity name
+     - `className`: `String` - Class name as specified by the user (usually the same as `name`)
+     - `isAbstract`: `Bool` - Whether the entity is an abstract or not
+     - `superEntity`: `String` - Name of the super (parent) entity
+     - `uniquenessConstraints`: `Array<Array<String>>` - List of uniqueness constraints, each being a list of
+       attributes
+     - `attributes`: `Array` - List of attributes (see `Entity Attribute`)
+     - `relationships`: `Array` - List of relationships (see `Entity Relationship`)
+     - `fetchedProperties`: `Array` - List of fetched properties (see `Entity Fetched Property`)
+   - `fetchRequests`: `Dictionary` - All fetch requests, grouped by entity. Each key will be an entity name, each value
```

Comment:
My OCD is at it again 😅
Seems not uniform to have `entities` and `fetchRequests` be dictionaries… but `attributes`/`relationships`/`fetchedProperties` be Arrays. Shouldn't they be dictionaries too?

Maybe not useful to have those 3 indexed by name in the current template, but might be for another custom one, not sure?


I'm not strict on this, could go either way, but I thought it was worth mentioning in case you want to uniformize.

Reply:
That would make it harder if we ever want to have array operators such as array + array. Although we could also add a `.values` accessor for dictionaries 🤷‍♂️ 

Do note that `fetchRequests` is quite different. `entities` is for finding an entity by name, `fetchRequests` gives a **list** of requests for a given entity name. One could argue that we should move it **into** the corresponding entity.

Reply:
Ah right. Nah, keep it roughly organized the same as the xcdatamodel format structure, FetchRequests are separate in the format, keep them separate in the context and let the template aggregate them like now then.